lib/Doctrine/ORM/Internal/Hydration/AbstractHydrator.php

```diff
@@ -190,6 +190,27 @@ public function toIterable(ResultStatement $stmt, ResultSetMapping $resultSetMap
         }
     }
 
+    /**
+     * Initiates a row-by-row hydration.
+     *
+     * @deprecated
+     *
+     * @psalm-param array<string, mixed> $hints
+     *
+     * @return Generator<int, mixed>
+     */
+    public function toIterable(ResultStatement $stmt, ResultSetMapping $resultSetMapping, array $hints = []): iterable
```

Comment:
It saddens me that I've had to deprecated that shiny new method again, just because I wanted to switch the type declaration from `ResultStatement` to `Result`. Without this change, we cannot achieve DBAL 3 compatibility because `ResultStatement` has been removed in 3.0.

Changig or removing the type declaration would cause custom hydrators that override this very method to break.

If you know a better way, please advise.

Reply:
It's indeed a bit sad but I'm unsure how commonplace it is to implement custom hydrators (although it's documented here: https://www.doctrine-project.org/projects/doctrine-orm/en/latest/reference/dql-doctrine-query-language.html#custom-hydration-modes). Maybe the impact is not that big?

Reply:
Plus, those custom hydrators would only break if they override `toIterable()`. If you think that this break would be tolerable, I can revert the deprecation. tbh, I feel like `toIterable()` should've been `final` anyway.

Reply:
Let's mark it as final with `@final` then? And hope the backlash is small, and document the small BC-break.

Reply:
Done!